app/main.py

```diff
@@ -1,3 +1,73 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    @property
+    def km(self) -> int | float:
+        return self._km
+
+    @km.setter
+    def km(self, value: int | float) -> None:
+        if not isinstance(value, (int | float)):
+            raise ValueError(
+                f"Unsupported type. <class 'float'> "
+                f"or <class 'int'> were expected. "
+                f"Got {type(value)} instead."
+            )
+        self._km = value
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: int | float | Distance) -> Distance:
+        if isinstance(other, (Distance, int, float)):
+            if isinstance(other, (int, float)):
+                other = Distance(other)
+            return Distance(self.km + other.km)
+
+    def __iadd__(self, other: int | float | Distance) -> Distance:
+        if isinstance(other, (Distance, int, float)):
+            if isinstance(other, (int, float)):
+                other = Distance(other)
+            self.km += other.km
+            return self
+
+    def __mul__(self, other: int | float) -> Distance:
+        if isinstance(other, (int, float)):
+            other = Distance(other)
+            return Distance(self.km * other.km)
+
+    def __truediv__(self, other: int | float) -> Distance:
+        if isinstance(other, (int, float)):
+            other = Distance(other)
+            return Distance(round(self.km / other.km, 2))
+
+    def __lt__(self, other: int | float | Distance) -> bool:
+        if isinstance(other, (Distance, int, float)):
+            if isinstance(other, (int, float)):
+                other = Distance(other)
+            return self.km < other.km
+
+    def __gt__(self, other: int | float | Distance) -> bool:
+        if isinstance(other, (Distance, int, float)):
+            if isinstance(other, (int, float)):
+                other = Distance(other)
+            return self.km > other.km
+
+    def __eq__(self, other: int | float | Distance) -> bool:
+        if isinstance(other, (Distance, int, float)):
+            if isinstance(other, (int, float)):
+                other = Distance(other)
+            return self.km == other.km
+
+    def __le__(self, other: int | float | Distance) -> bool:
```

Comment:
Here you compare self (instance of Distance class) with possible int or float. Also why do you use 'not' operator here?

Reply:
> Here you compare self (instance of Distance class) with possible int or float. Also why do you use 'not' operator here?
---Here you compare self (instance of Distance class) with possible int or float.
Actually I believe that's not true, I use ```return not self > other``` so it goes to ```__gt__``` method at this line and ```__gt__``` method is checking if other is ```int```, ```float``` or ```Distance```. If it's ```int``` or ```float``` it does ```other = Distance(other)``` under ```__gt__``` method as you can see. 
p.s. screenshot is attached with comparison of ```Distance``` with ```int/float/Distance```
![NVIDIA_Share_wStengcfP4](https://github.com/mate-academy/py-distance-class/assets/13889031/1a847189-5ef9-45d1-8f8e-9b0c40dd8c06)

---Also why do you use 'not' operator here?
Because when I use ```return not self > other``` in ```__le__``` ```less-equal``` magic method it returns ```True``` when ```self == other``` or ```self < other```, why should I use 2 comparisons when I can just write the key word ```not``` and just use ```self > other```? (Btw this was the reccomendation from @YuriiKindrat on our QnA I had: ```self == other``` or ```self < other``` before that.

Reply:
That's a good point and clever implementation. I totally overlooked that.
Also along with that you can use the following:
[https://docs.python.org/3/library/functools.html#functools.total_ordering](url)